### PR TITLE
Pre commit all files

### PR DESCRIPTION
**Pull request by ewels**
Fixed pre-commit tests on all modules.

Enabled pre-commit `--all-files` on GitHub Actions, so that this check will now always check everything.

This means that if someone pushes a bunch of commits in one go, it'll test all of them and not just the last in the chain.

### CODE REVIEW

**Review by maxulysse — APPROVED** [review]

File: modules/nf-core/kmcp/merge/meta.yml [hunk 1]
```diff
@@ -14,7 +14,7 @@ tools:
       documentation: "https://github.com/shenwei356/kmcp#documents"
       tool_dev_url: "https://github.com/shenwei356/kmcp"
       doi: "10.1093/bioinformatics/btac845"
-      licence: "['MIT']"
+      license: ["['MIT']"]
```
Comment (mahesh-panchal) [inline modules/nf-core/kmcp/merge/meta.yml:17]:
```suggestion
      license: ["MIT"]
```

File: subworkflows/nf-core/bam_variant_demix_boot_freyja/meta.yml [hunk 2]
```diff
@@ -22,10 +22,10 @@ input:
         Structure: [ val(meta), path(fasta) ]
         Groovy Map containing sample information e.g. [ id:'test', single_end:false ] and the fasta reference used for the sorted BAM file
   - val_repeats:
-      type: value (int)
+      type: string (int)
       description: Number of bootstrap repeats to perform
   - val_db_name:
-      type: value (string)
+      type: string (string)
```
Comment (mahesh-panchal) [inline subworkflows/nf-core/bam_variant_demix_boot_freyja/meta.yml:28]:
```suggestion
      type: string
```

**Review by mahesh-panchal — APPROVED** [review]
There's a few things like what's in the Licence strings that bothers me, but the autolinting is generally correct.

File: subworkflows/nf-core/bam_variant_demix_boot_freyja/meta.yml [hunk 3]
```diff
@@ -22,10 +22,10 @@ input:
         Structure: [ val(meta), path(fasta) ]
         Groovy Map containing sample information e.g. [ id:'test', single_end:false ] and the fasta reference used for the sorted BAM file
   - val_repeats:
-      type: value (int)
+      type: string (int)
```
[the remark this answers is not in the document]
Comment (ewels) [reply subworkflows/nf-core/bam_variant_demix_boot_freyja/meta.yml:25]: Yup, available types are:

```yaml
['map', 'file', 'directory', 'string', 'integer', 'float', 'boolean', 'list']
```

Weird that this didn't get picked up by the CI! `string (int)` is _not_ in that list..

File: modules/nf-core/khmer/uniquekmers/meta.yml [hunk 4]
```diff
@@ -18,7 +18,7 @@ input:
       description: fasta file
       pattern: "*.{fa,fasta}"
   - kmer_size:
-      type: value
+      type: string
```
[the remark this answers is not in the document]
Comment (ewels) [reply modules/nf-core/khmer/uniquekmers/meta.yml:21]: Yes, `integer` given the pattern below (which can be removed)

```suggestion
      type: integer
```

File: modules/nf-core/khmer/uniquekmers/meta.yml [hunk 5]
```diff
@@ -18,7 +18,7 @@ input:
...
       description: k-mer size to use
       pattern: "[0-9]+"
```
Comment (ewels) [inline modules/nf-core/khmer/uniquekmers/meta.yml:23]:
```suggestion
```

File: modules/nf-core/hmmer/eslalimask/meta.yml [hunk 6]
```diff
@@ -22,22 +22,22 @@ input:
       description: multiple sequence alignment, Stockholm or other formats
       pattern: "*"
   - fmask_rf:
-      type: val
+      type: string
```
[the remark this answers is not in the document]
Comment (ewels) [reply modules/nf-core/hmmer/eslalimask/meta.yml:25]: Ah I didn't check the `main.nf`, but looking there now yes I guess so  👍🏻

**Comment by ewels** [comment]
Yes, it'd be nice to make license one of an approved list to make them more homogenous. But I figured that we could take that at a later date, at least just start with having consistent meta YAML that we can parse.

**Comment by ewels** [comment]
Merging as all failures are unrelated to pre-commit. Will spin up issues about them so that they're not lost.

**Pull request merged**